crates/transaction/manifests/src/manifests_security_shield/manifests_securify_shield_securified_entity.rs

```diff
@@ -0,0 +1,113 @@
+#![allow(dead_code)]
+use crate::prelude::*;
+use std::ops::Deref;
+
+use profile_supporting_types::AnySecurifiedEntity;
+
+pub trait TransactionManifestSecurifySecurifiedEntity:
+    TransactionManifestSetRolaKey
+{
+    fn apply_security_shield_for_securified_entity(
+        securified_entity: AnySecurifiedEntity,
+        security_structure_of_factor_instances:
+        SecurityStructureOfFactorInstances,
+        apply_shield_manifest_kind: TransactionManifestApplySecurityShieldKind,
+    ) -> Result<TransactionManifest>;
+}
+
+impl TransactionManifestSecurifySecurifiedEntity for TransactionManifest {
+    /// Updates the security shield of a securified entity to `security_structure_of_factor_instances`.
+    ///
+    /// Also conditionally updates the Rola key of the entity - if it is new.
+    ///
+    /// Later once we have got a preview from Gateway - we will need to call:
+    /// * `modify_manifest_add_lock_fee_against_xrd_vault_of_access_controller`
+    ///
+    /// And when we know the fee we can calculate how much to top up the XRD vault of the AccessController
+    /// and call
+    /// * `modify_manifest_add_withdraw_of_xrd_for_access_controller_xrd_vault_top_up_paid_by_account`
+    ///
+    /// For timed confirmation - much later (`timed_recovery_delay_in_minutes` later ) the
+    /// host app will need to call `confirm_timed_recovery`
+    fn apply_security_shield_for_securified_entity(
+        securified_entity: AnySecurifiedEntity,
+        security_structure_of_factor_instances:
+        SecurityStructureOfFactorInstances,
+        apply_shield_manifest_kind: TransactionManifestApplySecurityShieldKind,
+    ) -> Result<Self> {
+        let kind = apply_shield_manifest_kind;
+        let entity_address = securified_entity.entity.address();
+
+        // ACCESS_CONTROLLER_CREATE_PROOF_IDENT
+        let mut builder = TransactionManifest::produce_owner_badge(
+            ScryptoTransactionManifestBuilder::new(),
+            &securified_entity.entity,
+        );
+
+        let access_controller_address = securified_entity
+            .securified_entity_control
+            .access_controller_address;
+
+        let factors_and_time_input = &AccessControllerFactorsAndTimeInput::new(
+            &security_structure_of_factor_instances,
+        );
+
+        // INITIATE RECOVERY
+        let (init_method, init_input) =
+            kind.input_for_initialization(factors_and_time_input);
+        builder = builder.call_method(
+            access_controller_address.scrypto(),
+            init_method,
+            (init_input.deref(),),
+        );
+
+        // QUICK CONFIRM RECOVERY - Only if we can exercise the confirmation role explicitly.
+        if let Some((confirm_method, confirm_input)) =
```

Comment:
Only call "quick confirm" if we can exercise confirmation role explicitly. Else we skip quick confirm and we will LATER (`number_of_days_until_timed_confirmation_is_callable` later) make another transaction.